lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2507,59 +2507,51 @@ struct SortedFuncList {
 
 ArrayRef<Decl *>
 EmittedMembersRequest::evaluate(Evaluator &evaluator,
-                                ClassDecl *CD) const {
-  auto &Context = CD->getASTContext();
+                                IterableDeclContext *idc) const {
+  auto dc = cast<DeclContext>(idc->getDecl());
+  auto &Context = dc->getASTContext();
   SmallVector<Decl *, 8> result;
 
-  if (!CD->getParentSourceFile()) {
-    auto members = CD->getMembers();
+  if (!dc->getParentSourceFile()) {
+    auto members = idc->getMembers();
     result.append(members.begin(), members.end());
     return Context.AllocateCopy(result);
   }
 
-  // We need to add implicit initializers because they
-  // affect vtable layout.
-  TypeChecker::addImplicitConstructors(CD);
+  auto nominal = dyn_cast<NominalTypeDecl>(idc);
 
-  auto forceConformance = [&](ProtocolDecl *protocol) {
-    auto ref = CD->getParentModule()->lookupConformance(
-        CD->getDeclaredInterfaceType(), protocol);
-    if (ref.isInvalid()) {
-      return;
-    }
+  if (nominal) {
+    // We need to add implicit initializers because they
+    // affect vtable layout.
+    TypeChecker::addImplicitConstructors(nominal);
+  }
 
-    auto conformance = ref.getConcrete();
-    if (conformance->getDeclContext() == CD &&
-        conformance->getState() == ProtocolConformanceState::Incomplete) {
+  // Force any derivable conformances in this context. This ensures that any
+  // synthesized members will approach in the member list.
+  for (auto conformance : idc->getLocalConformances()) {
```

Comment:
This is a behavior change. Is it okay to ignore conformances from other decl contexts e.g. extensions?

Reply:
Yeah, because they add members to the extension so we'll synthesize them when you call getSemanticMembers() on the extension. The one exception is that CodingKeys is always synthesized in the type itself because you can define Encodable and Decodable conformances in different extensions and we want them to share the same CodingKeys, but we handle that separately below.

Reply:
It should be, unless those conformances are going back and inserting declarations into the main nominal type declaration. (like apparently `CodingKeys` does). Perhaps I need to do an audit of all of the conformance-synthesis code...

Reply:
Okay, I did a quick audit. `CodingKeys` is weird but handled specifically.